airtime_mvc/application/models/Preference.php

```diff
@@ -498,7 +498,12 @@ public static function SetStationDescription($description)
 
     public static function GetStationDescription()
     {
-        return self::getValue("description");
+        $description = self::getValue("description");
+        if (!empty($description)) {
+            return $description;
+        } else {
+            return "Powered by Airtime Pro";
```

Comment:
Good translation frameworks give you the option to provide context or
disambiguation in your translation macros. I believe gettext does this if
you add the pipe symbol  | to your strings. We may need to be running
gettext with some extra argument as well.

On Wed, Jun 10, 2015 at 12:39 PM, Daniel James <notifications@github.com>
wrote:

> In airtime_mvc/application/models/Preference.php
> <https://github.com/sourcefabric/airtime/pull/116#discussion_r32137799>:
>
> > @@ -498,7 +498,12 @@ public static function SetStationDescription($description)
> >
> >      public static function GetStationDescription()
> >      {
> > -        return self::getValue("description");
> > +        $description = self::getValue("description");
> > +        if (!empty($description)) {
> > +            return $description;
> > +        } else {
> > +            return "Powered by Airtime Pro";
>
> Using a placeholder is better than nothing, although then the translator
> might need to be provided with some context. For example 'Powered by %' in
> the case of Airtime Pro is an idiom and might be difficult to translate,
> compared to say 'Powered by % lithium-ion batteries' where % is the brand
> of batteries.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sourcefabric/airtime/pull/116/files#r32137799>.
>



-- 
Albert Santoni
Airtime Lead Developer, Sourcefabric
albert.santoni@sourcefabric.org

www.sourcefabric.org

720 Bathurst St.
Suite 203
Toronto, ON / M5S 2R4 / Canada